btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -216,14 +216,19 @@ func (sew *StakingEventWatcher) fetchDelegations() {
 					unbondingOutput:       delegation.UnbondingOutput,
 				}
 
-				// if we already have this delegation, skip it
+				// if we already have this delegation, we still want to check if it has changed,
 				// we should track both verified and active status for unbonding
-				if sew.unbondingTracker.GetDelegation(delegation.StakingTx.TxHash()) == nil {
+				changed, exists := sew.unbondingTracker.HasDelegationChanged(delegation.StakingTx.TxHash(), del)
```

Comment:
the delegation might have changed from verified to active status, in order to track undonging, we need the activation height, and data in the tracker might be stale, so let's check this

Reply:
this was a bit confusing the checks `exists && changed` or `!exists`

If the delegation didn't exists, we still push it as a new delegation to the `unbondingDelegationChan`?

Reply:
Yes, we want to track it, but we also want to push it to the channel if it exits and has changed (e.g went from verified -> activated)